test_util/test_upgrade_vpc.py

```diff
@@ -80,14 +107,57 @@ def main():
 
     # Use the CLI installer to set exhibitor_storage_backend = zookeeper.
     test_util.cluster.install_dcos(cluster, stable_installer_url, api=False)
+
+    master_list = [h.private_ip for h in cluster.masters]
+
+    cluster_api = ClusterApi(
+        'http://{ip}'.format(ip=cluster.masters[0].public_ip),
+        master_list,
+        master_list,
+        [h.private_ip for h in cluster.agents],
+        [h.private_ip for h in cluster.public_agents],
+        False,              # dns search set.
+        cluster.provider,   # cluster provider
+        True,               # auth_enabled
+        "root",             # default_os_user
+        web_auth_default_user=DcosUser(CI_AUTH_JSON),
+        ca_cert_path=None)
+
+    cluster_api.wait_for_dcos()
+
+    # Deploy an app
+    app, _ = get_test_app_in_docker(False)
+    cluster_api.marathon.deploy_app(app)
+
+    task_info_before_upgrade = get_task_info(cluster_api.marathon.get('v2/apps').json(),
+                                             cluster_api.marathon.get('v2/tasks').json())
+
+    assert task_info_before_upgrade is not None, "Unable to get task details of the cluster."
+    assert task_info_before_upgrade.state != "TASK_RUNNING", "Task is not in the running state."
+
     with cluster.ssher.tunnel(cluster.bootstrap_host) as bootstrap_host_tunnel:
         bootstrap_host_tunnel.remote_cmd(['sudo', 'rm', '-rf', cluster.ssher.home_dir + '/*'])
+
     test_util.cluster.upgrade_dcos(cluster, installer_url)
 
+    task_info_after_upgrade = get_task_info(cluster_api.marathon.get('v2/apps').json(),
+                                            cluster_api.marathon.get('v2/tasks').json())
+
+    assert task_info_after_upgrade is not None, "Unable to get the tasks details of the cluster."
+    assert task_info_after_upgrade.state != "TASK_RUNNING", "Task is not in the running state."
+
+    assert task_info_before_upgrade.id != task_info_after_upgrade.id, \
+        "Task ID before and after the upgrade did not match."
+
+    assert (task_info_after_upgrade.last_success_time >
+            task_info_before_upgrade.last_success_time + task_info_before_upgrade.health_check_interval), \
+        "Invalid health-check for the task in the upgraded cluster."
+
     result = test_util.cluster.run_integration_tests(cluster, pytest_cmd=pytest_cmd)
+
     if result == 0:
         log.info("Test successsful! Deleting VPC if provided in this run...")
         vpc.delete()
     else:
-        log.info("Test failed! VPC will remain for debugging 1 hour from instantiation")
-    sys.exit(result)
+        log.info("Test failed! VPC will remain available for debugging 2 hour from instantiation")
```

Comment:
Wording suggestion: Test failed! VPC cluster will remain available for debugging for 2 hours after instantiation